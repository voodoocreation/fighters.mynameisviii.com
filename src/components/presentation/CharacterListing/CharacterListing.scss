@import "../../../scss/variables", "../../../scss/mixins";

.CharacterListing {
  flex-basis: 33.333333%;
  padding: $spacing-quarter;

  img {
    width: 100%;
  }

  @include screen-above($screen-xs-min) {
    flex-basis: 25%;
  }

  @include screen-above($screen-sm-min) {
    flex-basis: 20%;
  }
}
