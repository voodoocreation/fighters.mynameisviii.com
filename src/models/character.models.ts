import { DeepPartial } from "redux";

import { createSlugFromString } from "../helpers/dataTransformers";
import { IMoveset, moveset } from "./moveset.models";

export interface ICharacter {
  imageUrl: string;
  movesets: IMoveset[];
  name: string;
  slug: string;
}

export const character = (
  options: DeepPartial<ICharacter> = {}
): ICharacter => ({
  imageUrl: options.imageUrl || "",
  movesets: options.movesets ? options.movesets.map(moveset) : [],
  name: options.name || "",
  slug: options.slug || createSlugFromString(options.name || "")
});
