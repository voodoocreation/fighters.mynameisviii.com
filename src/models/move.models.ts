import { DeepPartial } from "redux";

export interface IMove {
  combo: string;
  conditions?: string;
  name: string;
}

export const move = (options: DeepPartial<IMove> = {}): IMove => ({
  combo: options.combo || "",
  conditions: options.conditions || undefined,
  name: options.name || ""
});
