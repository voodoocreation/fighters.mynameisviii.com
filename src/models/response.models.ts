export interface ISuccess<Data> {
  ok: true;
  data: Data;
}

export const success = <T>(data: T): ISuccess<T> => ({ ok: true, data });

export interface IFailure {
  ok: false;
  message: string;
}

export const failure = (message: string): IFailure => ({ ok: false, message });
