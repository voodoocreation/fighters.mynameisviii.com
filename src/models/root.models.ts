export * from "./character.models";
export * from "./error.models";
export * from "./game.models";
export * from "./move.models";
export * from "./moveset.models";
export * from "./response.models";
