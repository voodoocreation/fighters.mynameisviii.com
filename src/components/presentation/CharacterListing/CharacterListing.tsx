import * as React from "react";
import { InjectedIntlProps, injectIntl } from "react-intl";

import { ICharacter, IGame } from "../../../models/root.models";

import Link from "../Link/Link";

import "./CharacterListing.scss";

interface IProps extends ICharacter, InjectedIntlProps {
  game: IGame;
  onClick: () => void;
}

class CharacterListing extends React.Component<IProps> {
  public render() {
    const { imageUrl, game, name, slug } = this.props;
    const { formatMessage } = this.props.intl;

    return (
      <article className="CharacterListing">
        <Link href={`#${slug}`} onClick={this.props.onClick}>
          <img
            src={imageUrl}
            alt={formatMessage(
              { id: "VIEW_MOVES_FOR_CHARACTER_FROM_GAME" },
              { character: name, game: game.title }
            )}
          />
        </Link>
      </article>
    );
  }
}

const CharacterListingWrapped = injectIntl(CharacterListing);

export default CharacterListingWrapped;
