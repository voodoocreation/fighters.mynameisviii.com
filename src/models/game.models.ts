import { DeepPartial } from "redux";

import { createSlugFromString } from "../helpers/dataTransformers";
import { character, ICharacter } from "./character.models";

export interface IGame {
  imageUrl: string;
  slug: string;
  title: string;
  characters: ICharacter[];
}

export const game = (options: DeepPartial<IGame> = {}): IGame => ({
  characters: options.characters ? options.characters.map(character) : [],
  imageUrl: options.imageUrl || "",
  slug: options.slug || createSlugFromString(options.title || ""),
  title: options.title || ""
});
