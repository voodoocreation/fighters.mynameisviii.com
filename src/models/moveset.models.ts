import { DeepPartial } from "redux";

import { IMove, move } from "./move.models";

export interface IMoveset {
  moves: IMove[];
  name: string;
}

export const moveset = (options: DeepPartial<IMoveset> = {}): IMoveset => ({
  moves: options.moves ? options.moves.map(move) : [],
  name: options.name || "",
});
