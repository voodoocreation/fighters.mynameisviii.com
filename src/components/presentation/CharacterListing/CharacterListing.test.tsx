import { character, game } from "../../../models/root.models";
import WrapperWithIntl from "../../../utilities/WrapperWithIntl";
import CharacterListing from "./CharacterListing";

const testGame = game({
  slug: "game-title",
  title: "Game title",
});

const testCharacter = character({
  imageUrl: "imageUrl",
  name: "Character Name",
});

const component = new WrapperWithIntl(CharacterListing).withDefaultProps({
  ...testCharacter,
  game: testGame,
  onClick: jest.fn(),
});

describe("[presentation] <CharacterListing />", () => {
  const wrapper = component.mount();

  it("renders with the correct link href when unselected", () => {
    expect(wrapper.find("Link").prop("href")).toBe("#character-name");
  });

  describe("when clicking the listing", () => {
    it("clicks the link", () => {
      wrapper.find("Link").simulate("click");
    });

    it("calls the onClick prop", () => {
      expect(component.props.onClick).toHaveBeenCalledTimes(1);
    });
  });

  it("matches snapshot", () => {
    expect(wrapper.render()).toMatchSnapshot();
  });
});
