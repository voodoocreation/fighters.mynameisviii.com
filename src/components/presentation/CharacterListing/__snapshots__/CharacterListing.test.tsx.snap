// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[presentation] <CharacterListing /> matches snapshot 1`] = `
<article
  class="CharacterListing"
>
  <a
    href="#character-name"
  >
    <img
      alt="View moves for Character Name from Game title"
      src="imageUrl"
    />
  </a>
</article>
`;
